src/json/from_json.cr

```diff
@@ -85,7 +85,12 @@ end
                        } %}
   def {{type.id}}.new(pull : JSON::PullParser)
     location = pull.location
-    value = pull.read_int
+    value =
+      {% if type == "UInt64" %}
+        pull.read_raw
+      {% else %}
+        pull.read_int
+      {% end %}
     begin
       value.to_{{method.id}}
     rescue ex : OverflowError
```

Comment:
There is a deficiency here regarding exceptions.

I think all cases should raise `JSON::ParseException`, but with this `UInt64` is special.

```console
$ cr eval 'require "json"; UInt64.from_json("1" + UInt64::MAX.to_s)'

Unhandled exception: Invalid UInt64: 118446744073709551615 (ArgumentError)
  from src/string.cr:477:5 in 'to_u64'
  from src/json/from_json.cr:76:1 in 'new'
  from src/json/from_json.cr:13:3 in 'from_json'
  from /eval:1:17 in '__crystal_main'
  from src/crystal/main.cr:110:5 in 'main_user_code'
  from src/crystal/main.cr:96:7 in 'main'
  from src/crystal/main.cr:119:3 in 'main'
  from __libc_start_main
  from _start
  from ???

$ cr eval 'require "json"; Int64.from_json("1" + UInt64::MAX.to_s)' 

Unhandled exception: Invalid Int64: 118446744073709551615 at line 1, column 1 (JSON::ParseException)
  from src/json/token.cr:25:7 in 'int_value'
  from src/json/pull_parser.cr:84:5 in 'int_value'
  from src/json/pull_parser.cr:214:5 in 'read_int'
  from src/json/from_json.cr:76:1 in 'new'
  from src/json/from_json.cr:13:3 in 'from_json'
  from /eval:1:17 in '__crystal_main'
  from src/crystal/main.cr:110:5 in 'main_user_code'
  from src/crystal/main.cr:96:7 in 'main'
  from src/crystal/main.cr:119:3 in 'main'
  from __libc_start_main
  from _start
  from ???
```

This might be fixable with just this change?

```suggestion
    rescue ex : OverflowError | ArgumentError
```

Reply:
Specs that show that this exception is raised correctly would also be good to have.

Reply:
Yeah, I agree. All errors should be `JSON::ParseError`

Reply:
Nice catch, I updated the MR with a test case.